packages/eslint-plugin-pf-codemods/lib/rules/v5/table-rename-TableComposable.js

```diff
@@ -0,0 +1,92 @@
+const {renameComponents, getPackageImports} = require("../../helpers");
+
+// https://github.com/patternfly/patternfly-react/pull/8892
+module.exports = {
+  meta: {fixable: "code"},
+  create: function (context) {
+    const imports = getPackageImports(context, "@patternfly/react-table");
+    const tableComposableImport = imports.find(
+      (imp) => imp.imported.name === "TableComposable"
+    );
+
+    return imports.length === 0 || !tableComposableImport
+      ? {}
+      : {
+          // update opening/closing elements
+          JSXIdentifier(node) {
+            const nodeName = node.name;
+            const importedNode = imports.find(
+              (imp) => imp.local.name === nodeName
+            );
+            if (nodeName === "Table") {
+              // if data-codemods attribute, do nothing
+              const parentNode = node.parent;
+              const isOpeningTag = parentNode.type === "JSXOpeningElement";
+              const openingTagAttributes = isOpeningTag
+                ? parentNode.attributes
+                : parentNode.parent.openingElement.attributes;
+              const hasDataAttr =
+                openingTagAttributes &&
+                openingTagAttributes.filter(
+                  (attr) => attr.name?.name === "data-codemods"
+                ).length;
+              if (hasDataAttr) {
+                return;
+              }
+            }
+            if (
+              nodeName.includes("TableComposable") &&
+              importedNode.imported.name === importedNode.local.name // don't rename an aliased component
```

Comment:
This would catch examples where we might have `import { Table as TableComposable }`?

I notice we are only looking for nodeName = "TableComposable" or "Table", so if someone is using an alias in general (`Table as PFTable` for example) we won't catch that. Is that by design? I know there's  other stuff happening with the setup rule and such so just want to make sure I understand well enough.

Reply:
@jenny-s51 so I think I see what Dallas is talking about here. In the `test.tsx` file, if I import `TableComposable as TC`, the data-codemods attribute doesn't get applied on fix, so it'll eventually get its import path updated to deprecated as well. Using an alias, the `nodeName` will end up being TC so this if block will never be true.

From there, another issue is that the `imported name === local name` conditional is not applying the data attribute (as well as the `needsDataAttr` variable on line ~42). We'd want the data attribute applied even on an alias, but we could omit the rename if an alias exists.

Reply:
Thanks @gitdallas and @thatblindgeye for extra clarification - I've updated the logic here to account for cases when imports are aliased. If we are using an aliased import, the data attribute now gets applied on the aliased component, as Eric suggested. This prevents the import path from getting updated to deprecated